### PR TITLE
port toranj test23 multicast_traffic

### PR DESCRIPTION
**Pull request by priyankaCh22**
seeing failures in receiving multicast messages.
Please find attached logs.
[silk_multicast_traffic.log](https://github.com/openthread/silk/files/5449968/silk_multicast_traffic.log)

### CODE REVIEW

File: silk/tests/openthread/ot_test_multicast_traffic.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+    @testcase.test_method_decorator
+    def test02_verify_multicast_all_nodes(self):
+        #
+        #    r1 ---- r2 ---- r3 ---- r4
+        #            |               |
+        #            |               |
+        #           fed             sed
+        #
+        for node1, node2 in [(self.r1, self.r3), (self.r1, self.fed), (self.r3, self.fed),
```
Comment (priyankaCh22) [inline silk/tests/openthread/ot_test_multicast_traffic.py]: Added extra allowlist for the nodes as the test was failing saying unknownneighbor.
i.e.
[2020-10-26 23:10:13,619] [silk.Dev-5B97.wpantund] [DEBUG] wpantund[22295]: NCP => [0000000457737][INFO]-MAC-----: Frame rx failed, error:UnknownNeighbor

Adding this allowlisting resolved error:UnknownNeighbor.

File: silk/tests/openthread/ot_test_multicast_traffic.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+        self.send_mcast(self.r2, self.r2.ip6_lla, LL_ALL_NODES, [self.r1, self.r2, self.r3, self.fed],
+                        [self.r4, self.sed])
+
+        # r3 =>> mesh-local all-nodes.
+        self.send_mcast(self.r3, self.r3.ip6_mla, ML_ALL_NODES, [self.r1, self.r2, self.r3, self.r4, self.fed])
+
+    @testcase.test_method_decorator
+    def test03_verify_multicast_all_routers(self):
+        # r3 =>> link-local all-routers.
+        self.send_mcast(self.r3, self.r3.ip6_mla, LL_ALL_ROUTERS, [self.r2, self.r3, self.r4],
```
Comment (priyankaCh22) [inline silk/tests/openthread/ot_test_multicast_traffic.py]: seeing message not received error here

File: silk/tests/openthread/ot_test_multicast_traffic.py (hunk 1 above)
Comment (qilinkang) [reply silk/tests/openthread/ot_test_multicast_traffic.py]: Should not need to add extra allowlist. Can you please check the route table ?
>> Checked router table is correct. Issue is netcat does not support multicast beyond 1 hop raising failures with mesh multicasts.

File: silk/tests/openthread/ot_test_multicast_traffic.py (hunk 2 above)
Comment (qilinkang) [inline silk/tests/openthread/ot_test_multicast_traffic.py]: self.r3.ip6_lla ?

File: silk/tests/openthread/ot_test_multicast_traffic.py (hunk 1 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_multicast_traffic.py]: Ack

File: silk/tests/openthread/ot_test_multicast_traffic.py (hunk 2 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_multicast_traffic.py]: Ack

**Review by qilinkang — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**